text/0308-cli-advisories.md

```diff
@@ -0,0 +1,263 @@
+
+| Field                  | Description                                                                                                                                                | Format                                 | Mandatory? |
+| ---------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------- | -------------------------------------- | ---------- |
+| Summary                | A paragraph with more information                                                                                                                          | Free-form text                         | Yes        |
+| Affected component     | The CLI or the construct library (or both)                                                                                                                 | A non-empty subset of {CLI, Framework} | Yes        |
+| Affected version range | Version range using the semver format                                                                                                                      | semver                                 | No         |
+| Suggested fix          | An action the user can take to mitigate the problem. If it requires a longer explanation, this field should be omitted in favor of the resource linked to. | Free-form text                         | No         |
+
+In order to keep the issue human readable as well as easily parseable, each of
+these fields should be indicated by a Markdown header (`### Summary`, `###
+Affected component` etc). The piece of text following the header will be
+interpreted as the content of the field.
+
+#### GitHub workflow
+
+To be considered an advisory, a GitHub issue must be pinned and have the
+following tags: `p0` and a new tag, `advisory`. The `advisory` tag will work as
+an acknowledgement that author of the issue is making a conscious decision to
+publish an advisory that will be seen by potentially hundreds of thousands of
+users. Adding this tag will trigger a new workflow, which will inspect the issue
+and check whether its body contains the right data format (see **Data format**
+section). If the issue doesn’t comply with the specification, the workflow will
+post a message (similar to a CI report, for example).
+
+#### CLI logic
+
+The CLI will make unauthenticated requests to the GitHub Issues API and query
+for issues that match the tag system described in the **GitHub workflow**
+section. Then it will apply additional filters on the result:
+
+* Format filter: whether the body of the issue complies with the data format
+  specification.
+* Version filter: whether the version range contained in the issue matches the
+  CLI version or the framework version (depending on the affected component,
+  also present in the issue). To check which version of the framework is being
+  used, the CLI will read the cloud assembly (`tree.json` file, specifically).
```

Comment:
tree? is that where the version appears? Can we rely on the fact that there is a cloud assembly? There are some CLI commands that won't synthesize a cloud assembly I believe.

Reply:
Sorry, I asked if `tree.json` is where we find the version? Isn't that in `manifest.json` somewhere?

Reply:
Oh, my bad. The version in `tree.json` is of `constructs`. This information doesn't seem to be readily available anywhere, actually. We might need to add it to the metadata.

Reply:
There's a version handshake between the CLI and the framework that is used to verify that the cloud assembly is compatible. Maybe we can utilize this. See [RFC 110](https://github.com/aws/aws-cdk-rfcs/blob/master/text/00110-cli-framework-compatibility-strategy.md) by @iliapolo.

Reply:
Yes, but that is the version of the cloud assembly schema, which is too coarse-grained for this purpose.